This will disable FastMuxSwitch if a client is actively transcoding a liveTV stream.

jvl711 opened the pull request:
It appears that when SageTV is doing server side transcoding that it is not able to do a fastmuxswitch and have the client transition to the new program/timeslot.  I have noticed this behavior in the Placeshifter, and the andriod client.  With this change, placeshifter is able to transition between timeslots without issue.  And when you are using a client without transcoding, it does a fastmuxswitch without issue.

Narflex commented on java/sage/MiniPlayer.java:
```diff
@@ -617,6 +617,20 @@ void checkForByteBasedSeeking(java.io.File file) {
+   */
+  public boolean isTranscoding()
+  {
+    if(this.mpegSrc.getTranscoder() != null && serverSideTranscoding)
```
Remove the 'this.' you're using here and other places, it's not needed.

Narflex replied:
Check if mpegSrc is null before calling a method on it

Narflex commented on java/sage/Seeker.java:
```diff
@@ -4150,8 +4171,16 @@ else if (currRecMR != null && currRecMR.getSchedulingAiring().getEndTime() <= cu
+
+            switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 && es.currRecordFile.getSize() > 0;
+
+            if(this.isAClientControllingCaptureDevice(es.capDev) && this.isClientTranscodingMediaFile(es.currRecordFile))
```
Remove the 'this.' and add a space after the 'if'

Narflex commented on java/sage/Seeker.java:
```diff
@@ -4403,12 +4432,20 @@ else if (currRecords.contains(es.forceWatch) && (es.forceWatch != es.currRecord)
-              es.currRecordFile.getSize() > 0);
-          switchAny = true;
-        }
+	{
```
Fix the indentation of this block

Narflex commented on java/sage/Seeker.java:
```diff
@@ -4403,12 +4432,20 @@ else if (currRecords.contains(es.forceWatch) && (es.forceWatch != es.currRecord)
+	  
+	  switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 && es.currRecordFile.getSize() > 0;
+	  
+	  if(this.isAClientControllingCaptureDevice(es.capDev) && this.isClientTranscodingMediaFile(es.currRecordFile))
```
Same as before

Narflex commented on java/sage/VideoFrame.java:
```diff
@@ -4829,6 +4829,25 @@ public boolean isCaptureDeviceControlled(CaptureDevice capDev)
+  {    
+    if(player instanceof MiniPlayer)
+    {
+      //((MiniPlayer)player).getFi
```
Remove this line

Narflex commented on java/sage/VideoFrame.java:
```diff
@@ -4829,6 +4829,25 @@ public boolean isCaptureDeviceControlled(CaptureDevice capDev)
+    if(player instanceof MiniPlayer)
+    {
+      //((MiniPlayer)player).getFi
+      if(this.getCurrFile() == mf && ((MiniPlayer)player).isTranscoding())
```
Space after if, remove the 'this.'

jvl711 commented:
@Narflex 

I think I addressed all of the issues.  Sorry about the multiple pushes.  Some of the formatting issues were only in Git and not showing in my source.  I was taking a while for me to figure it out.

Thanks,
Josh

Narflex commented:
Thanks!
